File: tests/basic/inline_decl_typ.cpp
typedef unsigned int uint;

typedef const double cdouble;

typedef struct { uint x; uint y; } vect;

typedef vect myvect; // typedef to be removed

typedef int** mat2d;

typedef int*** mat3d; // typedef to be removed

cdouble f(uint n, myvect v, mat3d m) {
   return 2.0;
}

int main() {
  uint x;
  uint t[3];
  cdouble y1 = 1.0, t2 = 2.0;
  myvect v;
  mat2d m;
  mat3d M;
  mat3d* T;
}

File: tests/basic/inline_decl_typ.ml
open Optitrust
open Run
(* Not the expected result :
   - missing a "const" in the printing of type double;
   - mat3d typedef seems to vanish, why? *)

let _ =
    run
    ( fun _ ->
        set_init_source"inline_typ.cpp";
        Declaration.inline_typedef false ~decl_path:[cTypDef "uint"] ();
        Declaration.inline_typedef false ~decl_path:[cTypDef "cdouble"] ();
        Declaration.inline_decl false ~decl_path:[cTypDef "mat3d"] ();
        dump()
    )